sourcegraph/sourcegraph
# [Cody PLG] add mutation & gql field to upgrade users to Cody Pro

This is temporary for Dec GA and will be removed after Feb GA eventually. 
- Adds `cody_pro_enabled_at` column to `users` table. 
- Adds `cody_pro_enabled` and `cody_pro_enabled_at` gql fields on `type User`.
- Adds `upgrade_to_cody_pro` gql mutation.

## Test plan
```graphql
mutation {
  upgradeToCodyPro(user: "VXNlcjox") {
    id
    username
    codyProEnabled
    codyProEnabledAt
  }
}
```

- `sg start dotcom`
- Set feature flat `cody_pro_dec_ga` to true
- Performing the gql mutation should be successful
- Performing the mutation again should result in "user already...."
- Set the feature flag to false.
- Performing the mutation again should result in "feature not enabled"
